From 6c51eac0c738d3657e6b6c72f6bdf861a7774298 Mon Sep 17 00:00:00 2001
From: Thang Tran <thuutran@amperecomputing.com>
Date: Mon, 13 Sep 2021 20:24:09 +0700
Subject: [PATCH] add prioDbusSdrBase for fru custom handler priority

prioDbusSdrBase is set to be 11 that will only take over
prioOpenBmcBase.

The storagecommand handlers should take priority over
the handlers in storagehander if it is enabled by dynamic
sensors.

Change-Id: I6e9cffacfac4127d2307e8c5e548c84323742c94
Signed-off-by: Willy Tu <wltu@google.com>
---
 dbus-sdr/storagecommands.cpp | 6 +++---
 include/ipmid/api-types.hpp  | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/dbus-sdr/storagecommands.cpp b/dbus-sdr/storagecommands.cpp
index 5926615..cc88e26 100644
--- a/dbus-sdr/storagecommands.cpp
+++ b/dbus-sdr/storagecommands.cpp
@@ -1246,16 +1246,16 @@ void registerStorageFunctions()
     startMatch();
 
     // <Get FRU Inventory Area Info>
-    ipmi::registerHandler(ipmi::prioOemBase, ipmi::netFnStorage,
+    ipmi::registerHandler(ipmi::prioDbusSdrBase, ipmi::netFnStorage,
                           ipmi::storage::cmdGetFruInventoryAreaInfo,
                           ipmi::Privilege::User, ipmiStorageGetFruInvAreaInfo);
     // <READ FRU Data>
-    ipmi::registerHandler(ipmi::prioOpenBmcBase, ipmi::netFnStorage,
+    ipmi::registerHandler(ipmi::prioDbusSdrBase, ipmi::netFnStorage,
                           ipmi::storage::cmdReadFruData, ipmi::Privilege::User,
                           ipmiStorageReadFruData);
 
     // <WRITE FRU Data>
-    ipmi::registerHandler(ipmi::prioOpenBmcBase, ipmi::netFnStorage,
+    ipmi::registerHandler(ipmi::prioDbusSdrBase, ipmi::netFnStorage,
                           ipmi::storage::cmdWriteFruData,
                           ipmi::Privilege::Operator, ipmiStorageWriteFruData);
 
diff --git a/include/ipmid/api-types.hpp b/include/ipmid/api-types.hpp
index 518db03..61d9218 100644
--- a/include/ipmid/api-types.hpp
+++ b/include/ipmid/api-types.hpp
@@ -39,6 +39,7 @@ constexpr Group groupDCMI = 0xDC;
  * is being created at.
  */
 constexpr int prioOpenBmcBase = 10;
+constexpr int prioDbusSdrBase = 11;
 constexpr int prioOemBase = 20;
 constexpr int prioOdmBase = 30;
 constexpr int prioCustomBase = 40;
-- 
2.30.0

